packages/keys/src/private-key.ts

```diff
@@ -3,7 +3,7 @@ import { SOLANA_ERROR__KEYS__INVALID_PRIVATE_KEY_BYTE_LENGTH, SolanaError } from
 
 import { ED25519_ALGORITHM_IDENTIFIER } from './algorithm';
 
-function addPkcs8Header(bytes: ReadonlyUint8Array): ReadonlyUint8Array {
+function addPkcs8Header(bytes: ReadonlyUint8Array): ReadonlyUint8Array<ArrayBuffer> {
```

Comment:
Here, we're able to express that it's readonly, _and_ non-shared (because we just created it inline).